src/libs/Violations/ViolationsUtils.ts

```diff
@@ -0,0 +1,69 @@
+import reject from 'lodash/reject';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '@src/ONYXKEYS';
+import {PolicyCategories, PolicyTags, Transaction, TransactionViolation} from '@src/types/onyx';
+import possibleViolationsByField, {ViolationField} from './possibleViolationsByField';
+
+const ViolationsUtils = {
+    getViolationForField(transactionViolations: TransactionViolation[], field: ViolationField, translate: (key: string) => string): string[] {
+        return transactionViolations.filter((violation) => possibleViolationsByField[field]?.includes(violation.name)).map((violation) => translate(violation.name));
+    },
+
+    getViolationsOnyxData(
+        /** The transaction to check for policy violations. */
+        transaction: Transaction,
+        /** An array of existing transaction violations. */
+        transactionViolations: TransactionViolation[],
+        /** Indicates if the policy requires tags. */
+        policyRequiresTags: boolean,
+        /** Collection of policy tags and their enabled states. */
+        policyTags: PolicyTags,
+        /** Indicates if the policy requires categories. */
+        policyRequiresCategories: boolean,
+        /** Collection of policy categories and their enabled states. */
+        policyCategories: PolicyCategories,
+    ): {
+        onyxMethod: string;
+        key: string;
+        value: TransactionViolation[];
+    } {
+        let newTransactionViolations = [...transactionViolations];
+
+        if (policyRequiresCategories) {
+            const categoryViolationExists = transactionViolations.some((violation) => violation.name === 'categoryOutOfPolicy');
+            const categoryIsInPolicy = policyCategories[transaction.category]?.enabled;
```

Comment:
NAB but using vars prefixed with `is` and `has` makes it super obvious that they're bool
```suggestion
            const hasCategoryViolation = transactionViolations.some((violation) => violation.name === 'categoryOutOfPolicy');
            const isCategoryInPolicy = policyCategories[transaction.category]?.enabled;
```

Can/should we declare these two variables as bool? I'm not sure if it's possible for `enabled` to be absent, which would make the second var undefined, and it can be null if ` policyCategories[transaction.category]?` returns null